x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SlmHealthIndicatorService.java

```diff
@@ -87,25 +108,83 @@ public HealthIndicatorResult calculate(boolean explain) {
+                    new HealthIndicatorImpact(
+                        2,
+                        "Some automated snapshots have not had a successful execution recently. Indices restored from affected "
+                            + "snapshots may not contain recent changes.",
+                        List.of(ImpactArea.BACKUP)
+                    )
+                );
+
+                return createIndicator(
+                    YELLOW,
+                    "Encountered [" + unhealthyPolicies.size() + "] unhealthy snapshot lifecycle management policies.",
+                    createDetails(explain, unhealthyPolicies, slmMetadata),
+                    impacts,
+                    List.of(
+                        new UserAction(
+                            ACTION_CHECK_RECENTLY_FAILED_SNAPSHOTS,
+                            unhealthyPolicies.stream().map(SnapshotLifecyclePolicyMetadata::getName).toList()
+                        )
+                    )
+                );
+            }
+
             return createIndicator(
                 GREEN,
                 "Snapshot Lifecycle Management is running",
-                createDetails(explain, slmMetadata),
+                createDetails(explain, Collections.emptyList(), slmMetadata),
                 Collections.emptyList(),
                 Collections.emptyList()
             );
         }
     }
 
-    private static HealthIndicatorDetails createDetails(boolean explain, SnapshotLifecycleMetadata metadata) {
+    private boolean snapshotFailuresExceedWarningCount(SnapshotLifecyclePolicyMetadata policyMetadata) {
+        SnapshotInvocationRecord lastFailure = policyMetadata.getLastFailure();
+        if (lastFailure == null) {
+            // No failures yet to act on
+            return false;
+        }
+
+        // Determine if the most recent snapshot is a failure
+        SnapshotInvocationRecord lastSuccess = policyMetadata.getLastSuccess();
+        if (lastSuccess != null && lastSuccess.getSnapshotStartTimestamp() > lastFailure.getSnapshotStartTimestamp()) {
+            // Success was more recent than last failure
+            return false;
+        }
+
+        return policyMetadata.getInvocationsSinceLastSuccess() >= failedSnapshotWarnThreshold;
+    }
+
+    private static HealthIndicatorDetails createDetails(
+        boolean explain,
+        Collection<SnapshotLifecyclePolicyMetadata> unhealthyPolicies,
+        SnapshotLifecycleMetadata metadata
+    ) {
         if (explain) {
             return new SimpleHealthIndicatorDetails(
-                Map.of("slm_status", metadata.getOperationMode(), "policies", metadata.getSnapshotConfigurations().size())
+                Map.of(
+                    "slm_status",
+                    metadata.getOperationMode(),
+                    "policies",
+                    metadata.getSnapshotConfigurations().size(),
+                    "unhealthy_policies",
+                    unhealthyPolicies.size()
```

Comment:
Would it be useful to detail how many "missed invocations" we had for each policy we deem "unhealthy" ?

Reply:
I thought about it, but didn't want to overload the response size if there a lot of policies. That said, I imagine the order of magnitude of SLM policies is likely not ever that high in a system, so it's probably fine to add them in.